android/app/src/androidTest/java/PrnmTest.java

```diff
@@ -0,0 +1,304 @@
+     *      - proposes a payment channel to Bob
+     *  Bob:
+     *      - accepts the proposed channel from Alice
+     *  both (repeated 3 times):
+     *      - send 1 TX and wait for a TX from the other peer, Alice goes first
+     *      - exit without closing the channel
+     * @param s
+     * @throws Exception
+     */
+    public void runFirstTest(Setup s) throws Exception {
+        sharedSetup(s);
+        if (s.Index == 0) {
+            proposeChannel(s);
+            sendTx(false, 3);
+        }
+        else {
+            acceptChannel();
+            sendTx(true, 5);
+        }
+        assertOffChainBals(106, 94);
+    }
+
+    /**
+     * This function does the following:
+     *  both:
+     *      - sharedSetup
+     *      - restores the channel with persistence
+     *  both (repeated 3 times):
+     *      - send 1 TX and wait for a TX from the other peer, Bob goes first
+     *  Alice:
+     *      - asserts the final balances
+     *  Bob:
+     *      - finalizes the channel
+     *  both:
+     *      - settle the channel
+     * @param s
+     * @throws Exception
+     */
+    public void runSecondTest(Setup s) throws Exception {
+        sharedSetup(s);
+        if (s.Index == 0)
+            sendTx(true, 1);
+        else
+            sendTx(false, 2);
+        sharedTeardown(s);
+
+        assertOffChainBals(109, 91);
+        assertOnChainBals(1009, 991);
+    }
+
+    /**
+     * This function:
+     * - imports an account from the Setup
+     * - connects to the Ethereum node
+     * - creates a prnm client
+     * - adds the other peer to the client
+     * - starts the proposal, update and newChannel handlers
+     */
+    public void sharedSetup(Setup s) throws Exception {
+        this.setup = s;
+        Prnm.setLogLevel(4);    // INFO
+        prefix = "prnm-" + s.Aliases[s.Index];   // android log prefix
+        String appDir = ApplicationProvider.getApplicationContext().getFilesDir().getAbsolutePath();
+        String ksPath = appDir + "/keystore";
+        String dbPath = appDir + "/database";
+
+        Wallet wallet = Prnm.newWallet(ksPath, "0123456789");
+        Address onChain = wallet.importAccount(s.SKs[s.Index]);
+
+        String ethUrl = "ws://10.5.0.9:8545";
+        Config cfg = new Config(s.Aliases[s.Index], onChain, s.Adjudicator, s.Assetholder, ethUrl, "0.0.0.0", 5750);
+        client = new Client(ctx, cfg, wallet);
+
+        client.addPeer(s.Addresses[1-s.Index], s.Hosts[1-s.Index], s.Ports[1-s.Index]);
+        client.onNewChannel(this);
+        new Thread(() -> {
+            client.handle(this, this);
+        }).start();
+
+        client.enablePersistence(dbPath);
+        client.restore(ctx);
+    }
+
+    public void proposeChannel(Setup s) throws Exception {
+        Thread.sleep(5000);
+        BigInts initBals = Prnm.newBalances(eth(100), eth(100));
+        log("Opening Channel…");
+        client.proposeChannel(ctx, s.Addresses[1-s.Index], 60, initBals);
```

Comment:
I think it is recommended to access static members directly via the class rather than using a specific instance, so `Setup.Addresses[1-s.Index]` would make more sense.

Reply:
Yes, i just changed it to static yesterday.